lib/sqlalchemy/dialects/postgresql/psycopg2.py

```diff
@@ -551,13 +590,55 @@ def _log_notices(self, cursor):
 
 
 class PGCompiler_psycopg2(PGCompiler):
-    pass
+    def __init__(
+        self, dialect, statement, column_keys=None, inline=False, **kwargs
+    ):
+        self.dialect = dialect
+        self.multiple_rows = inline
+        self.execute_values_insert_template = None
+
+        super(
+            PGCompiler_psycopg2,
+            PGCompiler_psycopg2).__init__(
+            self,
+            dialect,
+            statement,
+            column_keys,
+            inline,
+            **kwargs)
+
+    # Override SQLCompiler.generate_values_placeholders_str - enable use of
+    # psycopg2 execute_values()
+    def generate_values_placeholders_str(
+            self, crud_params, returning_clause_exists):
+        # Currently not using psycopg2.execute_values() when there's a returning clause; need to add support
+        # for receiving multiple return values from insert query
+        if self.multiple_rows and not returning_clause_exists and self.dialect.psycopg2_execution_mode == EnumPsycopg2ExecutionMode.VALUES_BATCH:
+            self.execute_values_insert_template = "(" + \
+                ", ".join([c[1] for c in crud_params]) + ")"
+            return " VALUES %s"
+        else:
+            return super(
+                PGCompiler_psycopg2,
+                PGCompiler_psycopg2).generate_values_placeholders_str(
+                self,
+                crud_params,
+                returning_clause_exists)
 
 
 class PGIdentifierPreparer_psycopg2(PGIdentifierPreparer):
     pass
 
 
+class EnumPsycopg2ExecutionMode(Enum):
+    """
+    Enum type to describe psycopg2 execution mode of multiple parameters queries
+    """
+    SINGLE_STATEMENT = 'single_statement'
```

Comment:
Can we use a more descriptive set of names here ?    I'm not sure what "single statement" describes without reading the source code.   In all three cases, we are invoking a "statement", and we are calling a method on psycopg2 only once, so from that perspective, they are all "single statement".  Within the internals of psycopg2 executemany(), the default implementation runs the given statement for each parameter set, so that's also not "single".

I think a set of parameters / names that are more descriptive might be:   "executemany_mode = (EXECUTEMANY_DEFAULT|EXECUTEMANY_BATCH|EXECUTEMANY_VALUES)".   that way they are all specific to executemany and they indicate which ones are special and also map directly to the names used by psycopg2; someone using this API will likely be referring to psycopg2s documentation.

Reply:
ok